### ISSUE TITLE
Selectable add support for begin(), end() widget methods

### ISSUE THREAD

**Issue by fdezmc** [yiisoft/yii2-jui#46]
New feature to allow to make selectable any item, between the begin() and end() widget calls, matching the filter attribute value selector.

Uses:

```php
Selectable::begin([
     'clientOptions' => [
         'filter' => 'my-selectable-item'
         'tolerance' => 'fit',
     ],
 ]);
 ```
```html
<ul>
      <li class="my-selectable-item">Item 1</li>
      <li class="my-selectable-item">Item 2</li>
      <li class="no-selectable-item">Item 3</li>
      <li class="my-selectable-item">Item 4</li>
 </ul>
 <div>
      <div>
          <div class="my-selectable-item">Another item </div>
      </div>
 </div>
```

 ```php
 Selectable::end();
 ```

**Comment by fdezmc** [comment]
Hi @cebe my commit solve this issue and pass all test. Is more a bug than a enhancement because the Selectable Widget methods begin() and end() don't works as expected in the current version. I've extended the Selectable Widget with my solution for some projects but would be nice to have this functionality included by default in Yii2.

**Comment by fdezmc** [comment]
Hi @cebe this issue can be closed as much as the pull-request has been merged at  #47

**Comment by cebe** [comment]
#47 has not been merged yet.
